packages/server/src/open-feature.ts

```diff
@@ -71,22 +73,107 @@ export class OpenFeatureAPI extends OpenFeatureCommonAPI {
     return this;
   }
 
-  setProvider(provider: Provider): this {
-    // ignore no-ops
-    if (this._provider !== provider) {
+  /**
+   * Sets the provider that OpenFeature will use for flag evaluations of clients without a name.
+   * Setting a provider supersedes the current provider used in new and existing clients without a name.
+   *
+   * @param {Provider} provider The provider responsible for flag evaluations.
+   * @returns {GlobalApi} OpenFeature API
+   */
+  setProvider(provider: Provider): this;
+  /**
+   * Sets the provider that OpenFeature will use for flag evaluations of clients with the given name.
+   * Setting a provider supersedes the current provider used in new and existing clients with that name.
+   *
+   * @param {string} client The name to identify the client
+   * @param {Provider} provider The provider responsible for flag evaluations.
+   * @returns {GlobalApi} OpenFeature API
+   */
+  setProvider(client: string, provider: Provider): this;
```

Comment:
```suggestion
  setProvider(clientName: string, provider: Provider): this;
```

Same suggestion as in the client sdk.